chainntnfs/test_utils.go

```diff
@@ -208,6 +209,14 @@ func NewBitcoindBackend(t *testing.T, minerAddr string, txindex,
 	zmqBlockHost := "ipc:///" + tempBitcoindDir + "/blocks.socket"
 	zmqTxHost := "ipc:///" + tempBitcoindDir + "/tx.socket"
 
+	// since MacOS can't address domain sockets using this convention, to
+	// allow for running the unit tests on a local Mac environment, we opt
+	// to connect to a free local tcp port.
+	if runtime.GOOS == "darwin" {
```

Comment:
New versions of `bitcoind` don't support unix domain sockets anymore, so this isn't just a MacOS thing. Proper fix is here: https://github.com/lightningnetwork/lnd/pull/8273
Just need to find some time to find and fix the remaining unit test failures